avocado/utils/asset.py

```diff
@@ -105,91 +262,34 @@ def _get_writable_cache_dir(self):
-        :return: asset file if exists or None
-        :rtype: str or None
-        """
-        for cache_dir in self.cache_dirs:
-            cache_dir = os.path.expanduser(cache_dir)
-            asset_file = os.path.join(cache_dir, relative_path)
-
-            # To use a cached file, it must:
-            # - Exists.
-            # - Be valid (not expired).
-            # - Be verified (hash check).
-            if (os.path.isfile(asset_file) and
-                    not self._is_expired(asset_file, self.expire)):
-                try:
-                    with FileLock(asset_file, 30):
-                        if self._verify(asset_file):
-                            return asset_file
-                except Exception:  # pylint: disable=W0703
-                    exc_type, exc_value = sys.exc_info()[:2]
-                    log.error('%s: %s', exc_type.__name__, exc_value)
-        return None
+        if expire is None:
+            return False
+        creation_time = os.lstat(asset_path)[stat.ST_CTIME]
+        expire_time = creation_time + expire
+        if time.time() > expire_time:
+            return True
+        return False
 
-    def _create_metadata_file(self, asset_file):
-        """
-        Creates JSON file with metadata.
-        The file will be saved as "asset_file"_metadata.json
-        :param asset_file: The asset whose metadata will be saved
-        :type asset_file: str
+    def _verify(self, asset_path):
         """
-        if self.metadata is not None:
-            basename = os.path.splitext(asset_file)[0]
-            metadata_file = "%s_metadata.json" % basename
-            metadata = json.dumps(self.metadata)
-            with open(metadata_file, "w") as f:
-                f.write(metadata)
+        Verify if the `asset_path` hash matches the hash in the hash file.
 
-    def get_metadata(self):
-        """
-        Returns metadata of the asset if it exists or None.
-        :return: metadata
+        :param asset_path: full path of the asset file.
+        :returns: True when self.asset_hash is None or when it has the same
+        value as the hash of the asset_file, otherwise return False.
+        :rtype: bool
         """
-        parsed_url = urlparse.urlparse(self.name)
-        basename = os.path.basename(parsed_url.path)
-        cache_relative_dir = self._get_relative_dir(parsed_url)
-        asset_file = self._find_asset_file(os.path.join(cache_relative_dir,
-                                                        basename))
-        if asset_file is not None:
-            basename = os.path.splitext(asset_file)[0]
-            metadata_file = "%s_metadata.json" % basename
-            if os.path.isfile(metadata_file):
-                with open(metadata_file, "r") as f:
-                    metadata = json.loads(f.read())
-                    return metadata
-        return None
+        return bool(not self.asset_hash or
+                    (self.asset_hash and
```

Comment:
If not `not self.asset_hash` then `self.asset_hash` exists and you don't need to check again.

Reply:
Sorry to take so long to answer your comment. Here is the use case where the actual code is needed:

If the asset hash does not exist, the method needs to return True. The term where it needs to be verified is the following `not False or (False and False)`. It means the `asset_hash` does not exist, so the first and second part of the _or_ will return `False`, but the method needs to return `True` here, making the first part of the _or_ necessary.